src/java.base/share/classes/java/util/Objects.java

```diff
@@ -189,6 +189,66 @@ public static String toIdentityString(Object o) {
         return o.getClass().getName() + "@" + Integer.toHexString(System.identityHashCode(o));
     }
 
+    /**
+     * Checks that the specified object reference is an identity object.
+     *
+     * @param obj the object reference to check for identity
+     * @param <T> the type of the reference
+     * @return {@code obj} if {@code obj} is an identity object
+     * @throws NullPointerException if {@code obj} is {@code null}
+     * @throws IdentityException if {@code obj} is not an identity object
+     * @since Valhalla
+     */
+    @ForceInline
+    public static <T> T requireIdentity(T obj) {
+        Objects.requireNonNull(obj);
+        var cl = obj.getClass();
+        if (cl.isValue())
```

Comment:
```suggestion
        if (obj.getClass().isValue())
```

Same as the implementation of other variants of the `requireIdentity` method.

Reply:
`cl` is used to throw the IdentityException in the following line.

Reply:
okay.  thanks